app/controllers/foreman_deployments/api/v2/deployments_controller.rb

```diff
@@ -0,0 +1,73 @@
+module ForemanDeployments
+  module Api
+    module V2
+      class DeploymentsController < BaseController
+        before_filter :find_resource, :only => [:show, :configure, :configuration, :run]
+        before_filter :parse_stack, :only => [:show, :configure, :configuration, :run]
+
+        def_param_group :deployment do
+          param :deployment, Hash, :required => true, :action_aware => true do
+            param :name, String, :required => true, :desc => N_('Name for the deployment')
+            param :stack_id, :identifier, :required => true, :desc => N_('Id of the stack to deploy')
+          end
+        end
+
+        api :POST, '/deployments/', N_('Create a deployment')
+        param_group :deployment, :as => :create
+        def create
+          deployment_params = params[:deployment]
+
+          stack = ForemanDeployments::Stack.find(deployment_params.delete(:stack_id)) if deployment_params[:stack_id]
+          deployment_params[:configuration] = ForemanDeployments::Configuration.new(:stack => stack)
+
+          @deployment = ForemanDeployments::Deployment.new(deployment_params)
+          process_response @deployment.save
+        end
+
+        api :GET, '/deployments/', N_('List deployments')
+        def index
+          @deployments = resource_scope
+        end
+
+        api :PUT, '/deployments/:id/configuration/', N_('Configure a deployment')
+        param :id, :identifier, :required => true
+        param :values, Hash, :required => true,
+                             :desc => N_('Hash with configuration for tasks (Format: task ID => configuration values).')
+        def configuration
+          configuration_update = ForemanDeployments::Configuration.new(:values => params[:values])
+
+          ForemanDeployments::Config::LoadVisitor.load(@stack_definition, @deployment.configuration)
+          ForemanDeployments::Config::LoadVisitor.load(@stack_definition, configuration_update)
```

Comment:
Here we assume that multiple calls to `Task.configure` actually are adding configuration values, and not replacing them.
I prefer to keep this logic in the config storage, rather than give it to decision of the task creator.
Think about this use case:

``` Ruby
task.configure(:a => 'my val')
# config: { :a => 'my val' }
task.configure(:b => 'other val')
# config: { :a => 'my val', :b => 'other val' }
task.configure( :a => 'updated val', :c => 'something new' )
# config: { :a => 'updated val', :b => 'other val', :c => 'something new' }
```
That means that there is no way to remove `:b` completely (I don't want to add nil == empty constraint to the task if it's not necessary). Besides this, I think it's a bit confusing that `#configure` has side effects.
IMHO, if we want to give the task opportunity to deal with configuration merge, we should do this implicitly by adding another method for the task (something like `#merge_configuration`), and leave the `#configure` method to be a full update.